src/mobu/services/business/nublado.py

```diff
@@ -202,13 +242,36 @@ async def idle(self) -> None:
 
     async def hub_login(self) -> None:
         self.logger.info("Logging in to hub")
-        with self.timings.start("hub_login"):
-            await self._client.auth_to_hub()
-
-    async def spawn_lab(self) -> bool:
+        with self.timings.start("hub_login", self.annotations()) as sw:
+            try:
+                await self._client.auth_to_hub()
+            except ClientJupyterProtocolError as exc:
+                raise JupyterProtocolError.from_client_exception(
+                    exc,
+                    event=sw.event,
+                    annotations=sw.annotations,
+                    started_at=sw.start_time,
+                ) from exc
```

Comment:
Is this adding new info, the event, annotations, and start time, to what was in this exception before this change?  Or was that somehow getting in there before?

Reply:
Well, the problem is that there was no Nublado Client distinct from mobu, and hence no client exceptions, and factoring the two layers apart got messy.

Reply:
I *think* that for all of these exceptions, we will likely already be supplying a start time, but since we have it as part of the stopwatch there's no harm in specifying it.